src/init/starship.bash

```diff
@@ -58,8 +58,14 @@ starship_precmd() {
         STARSHIP_DURATION=$((STARSHIP_END_TIME - STARSHIP_START_TIME))
         PS1="$(::STARSHIP:: prompt --terminal-width="$COLUMNS" --status=$STARSHIP_CMD_STATUS --pipestatus="${STARSHIP_PIPE_STATUS[*]}" --jobs="$NUM_JOBS" --cmd-duration=$STARSHIP_DURATION)"
         unset STARSHIP_START_TIME
+        if [[ ${BLE_ATTACHED-} ]]; then
+            bleopt prompt_rps1="${PS1//[!$'\n']}$(::STARSHIP:: prompt --right --terminal-width="$COLUMNS" --status=$STARSHIP_CMD_STATUS --pipestatus="${STARSHIP_PIPE_STATUS[*]}" --jobs="$NUM_JOBS" --cmd-duration=$STARSHIP_DURATION)"
```

Comment:
Thanks for the updates! Could you check here again?

When the user sets `shopt -u extquote`, `"${PS1//[!$'\n']}"` (which is quoted inside `"..."`) fails to properly remove the characters with Bash <= 5.1 (I think Bash-5.2's behavior is a bug). It leaves `$`, `'`, `\`, and `n` and removes everything other than them.

An easy way is to use variable assignment such as `nlns=${...}` because the right-hand side of the assignments is not subject to pathname expansions and word splitting so does not need to be quoted (unless it contains the shell special delimiters like `;|&<>`).

```suggestion
            local nlns=${PS1//[!$'\n']}
            bleopt prompt_rps1="$nlns$(::STARSHIP:: prompt --right --terminal-width="$COLUMNS" --status=$STARSHIP_CMD_STATUS --pipestatus="${STARSHIP_PIPE_STATUS[*]}" --jobs="$NUM_JOBS" --cmd-duration=$STARSHIP_DURATION)"
```

If you don't like it, maybe there is another approach, e.g., to save and restore the state of `shopt -u extquote`, etc.

Reply:
Sure, no problem, I have used the `nlns` variable now.